lumicks/pylake/population/dwelltime.py

```diff
@@ -544,8 +583,100 @@ def exponential_mixture_log_likelihood(params, t, min_observation_time, max_obse
     return -np.sum(log_likelihood)
 
 
+def _handle_amplitude_constraint(
+    n_components, params, fixed_param_mask
+) -> Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]:
+    """Determines how many amplitudes actually need fitting.
+
+    For a single-component model the amplitude is 1 by definition. For an N-component model, where
+    N - 1 amplitudes are fixed, the free amplitude is determined by the constraint. For models that
+    have more than one free amplitude, the amplitudes are constrained to sum to 1.
+
+    Parameters
+    ----------
+    n_components : int
+        number of components in the mixture model
+    params : array_like
+        model parameters
+    fixed_param_mask : array_like
+        logical mask of fixed parameters
+
+    Returns
+    -------
+    Tuple[np.ndarray, Union[Dict, Tuple], np.ndarray]
+        This function returns the mask of parameters to be fitted, amplitude constraint function
+        and updated parameter vector (forced to be consistent with the constraint). It was split
+        out from `_exponential_mle_optimize` to reduce cognitive load.
+
+    Raises
+    ------
+    ValueError
+        If the sum of the provided fixed amplitudes exceeds 1.
+    ValueError
+        If all amplitudes are fixed but the amplitudes do not sum to 1.
+    """
+    if fixed_param_mask is None:
+        fixed_param_mask = np.zeros(params.shape, dtype=bool)
+    else:
+        if len(fixed_param_mask) != len(params):
+            raise ValueError(
+                f"Length of fixed parameter mask ({len(fixed_param_mask)}) is not equal to the "
+                f"number of model parameters ({len(params)})"
+            )
```

Comment:
dunno why I didn't notice this before, but couldn't we just do `elif`?
```suggestion
    if fixed_param_mask is None:
        fixed_param_mask = np.zeros(params.shape, dtype=bool)
    elif len(fixed_param_mask) != len(params):
        raise ValueError(
            f"Length of fixed parameter mask ({len(fixed_param_mask)}) is not equal to the "
            f"number of model parameters ({len(params)})"
        )
```

Reply:
You are right, but also:
![image](https://user-images.githubusercontent.com/19836026/231500665-9ff9eb5c-9ae1-4cc6-9ba4-6f75452a0bb3.png)